src/components/suggest_item/suggest_item.js

```diff
@@ -65,7 +63,10 @@ EuiSuggestItem.propTypes = {
   /**
    * Takes 'icon' for EuiIcon and 'color'. 'color' can be either our palette colors (primary, secondary, etc) or a hex value.
    */
-  type: PropTypes.object,
+  type: PropTypes.shape({
+    icon: IconPropType,
+    color: PropTypes.oneOfType([PropTypes.oneOf(COLORS), PropTypes.string]),
+  }).isRequired,
```

Comment:
Being more specific about what `type` wants, and making it required so it's always available. We can provide default values if that seems more like how people will use it.